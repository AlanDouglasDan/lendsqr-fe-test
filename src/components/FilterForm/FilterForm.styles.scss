@use "variables";

.form-label {
  color: variables.$dim-grey;
  font-weight: 500;
  font-size: 14px;
}

.btn-outline {
  border: 1px solid variables.$dim-grey;
  border-radius: 8px;
  font-size: 14px;
  font-weight: 600;
  color: variables.$dim-grey;
  height: 40px;
  border: none;
}

.btn-themed {
  background-color: variables.$green;
  border-radius: 8px;
  font-size: 14px;
  font-weight: 600;
  color: variables.$white;
  height: 40px;
  border: none;
}

.btn-themed:hover {
  background-color: variables.$green2;
}

.btn-outline:hover {
  background-color: variables.$light-grey;
}

.grid-2 {
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 14px;
}

.dropdown .form-control:focus,
.dropdown .form-select:focus {
  color: variables.$dim-grey;
  background-color: variables.$white;
  border-color: #9fe7e6;
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(57, 205, 204, 0.25);
}

.dropdown-toggle::after {
  display: none;
}

.form-dropdown {
  width: 270px !important;
  top: 25rem;
  left: calc(60px + 0.95rem);
}
