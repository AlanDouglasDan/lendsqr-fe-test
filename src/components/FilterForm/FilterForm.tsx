import { FilterInterface } from "types/user.types";
import "./FilterForm.styles.scss";

export interface Props {
  setFilters: (filters: any) => void;
  filters: FilterInterface;
  _organizations: String[];
  runFilterFunction: () => void;
  resetFilterFunction: () => void;
}

const FilterForm: React.FC<Props> = ({
  setFilters,
  filters,
  _organizations,
  runFilterFunction,
  resetFilterFunction,
}) => {
  return (
    <div className="dropdown-menu p-4 form-dropdown">
      <div className="mb-3">
        <label htmlFor="exampleDropdownFormOrganization" className="form-label">
          Organization
        </label>
        <select
          className="form-select mb-3 form-label"
          aria-label="Default select example"
          style={{ fontSize: "1rem" }}
          onChange={(e) => {
            setFilters({
              ...filters,
              organization: e.target.value,
            });
          }}
          value={filters.organization}
        >
          <option defaultValue="">Select</option>
          {_organizations?.map((item: any, index: any) => {
            return (
              <option value={item} key={index}>
                {item}
              </option>
            );
          })}
        </select>

        <label htmlFor="exampleDropdownFormUsername" className="form-label">
          Username
        </label>
        <input
          type="text"
          className="form-control"
          id="exampleDropdownFormUsername"
          placeholder="Username"
          onChange={(e) => {
            setFilters({
              ...filters,
              username: e.target.value,
            });
          }}
          value={filters.username}
        />
      </div>

      <div className="mb-3">
        <label htmlFor="exampleDropdownFormEmail2" className="form-label">
          Email
        </label>
        <input
          type="email"
          className="form-control"
          id="exampleDropdownFormEmail2"
          placeholder="Email"
          onChange={(e) => {
            setFilters({
              ...filters,
              email: e.target.value,
            });
          }}
          value={filters.email}
        />
      </div>

      <div className="mb-3">
        <label htmlFor="datepicker" className="form-label">
          Date
        </label>
        <input
          type="date"
          className="form-control"
          id="datepicker"
          onChange={(e) => {
            setFilters({
              ...filters,
              date: e.target.value,
            });
          }}
          value={filters.date}
        />
      </div>

      <div className="mb-3">
        <label htmlFor="exampleDropdownFormPhoneNumber" className="form-label">
          Phone Number
        </label>
        <input
          type="text"
          className="form-control"
          id="exampleDropdownFormPhoneNumber"
          placeholder="Phone Number"
          onChange={(e) => {
            setFilters({
              ...filters,
              phoneNumber: e.target.value,
            });
          }}
          value={filters.phoneNumber}
        />
      </div>

      <label htmlFor="exampleDropdownFormStatus" className="form-label">
        Status
      </label>

      <select
        className="form-select mb-3 form-label"
        aria-label="Default select example"
        style={{ fontSize: "1rem" }}
        onChange={(e) => {
          setFilters({
            ...filters,
            status: e.target.value,
          });
        }}
        value={filters.status}
      >
        <option defaultValue="pending">Pending</option>
        <option value="active">Active</option>
        <option value="inactive">Inactive</option>
        <option value="blacklisted">Blacklisted</option>
      </select>

      <div className="grid-2 mt-4">
        <button
          className="btn btn-outline w-100"
          onClick={() => resetFilterFunction()}
        >
          Reset
        </button>
        <button
          className="btn btn-themed w-100"
          onClick={() => runFilterFunction()}
        >
          Filter
        </button>
      </div>
    </div>
  );
};

export default FilterForm;
