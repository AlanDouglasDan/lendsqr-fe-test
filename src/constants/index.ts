import { UserInterface } from "../types/user.types";

enum API_URLS {
  GET_USERS = `https://6270020422c706a0ae70b72c.mockapi.io/lendsqr/api/v1/users`,
}

const TABLE_HEADERS: String[] = [
  "Organization",
  "Username",
  "Email",
  "Phone Number",
  "Date Joined",
  "Status",
];

const OPTIONS: Intl.DateTimeFormatOptions = {
  year: "numeric",
  month: "short",
  day: "numeric",
  hour: "numeric",
  minute: "numeric",
  hour12: true,
};

const DUMMY_USER_DATA: UserInterface = {
  id: "",
  createdAt: "",
  orgName: "",
  userName: "",
  email: "",
  phoneNumber: "",
  lastActiveDate: "",
  profile: {
    firstName: "",
    lastName: "",
    avatar: "",
    phoneNumber: "",
    gender: "",
    bvn: "",
    address: "",
    currency: "",
  },
  guarantor: {
    firstName: "",
    lastName: "",
    phoneNumber: "",
    gender: "",
    address: "",
  },
  accountBalance: "",
  accountNumber: "",
  socials: {
    facebook: "",
    twitter: "",
    instagram: "",
  },
  education: {
    level: "",
    employmentStatus: "",
    sector: "",
    duration: "",
    officeEmail: "",
    monthlyIncome: [],
    loanRepayment: "",
  },
};

const DUMMY_FILTER_DATA = {
  organization: "",
  username: "",
  email: "",
  phoneNumber: "",
  date: "",
  status: "",
};

const DUMMY_ORGANIZATION_DATA = [...Array(100)].map(
  (organization) => organization
);

export {
  API_URLS,
  TABLE_HEADERS,
  OPTIONS,
  DUMMY_USER_DATA,
  DUMMY_FILTER_DATA,
  DUMMY_ORGANIZATION_DATA,
};
