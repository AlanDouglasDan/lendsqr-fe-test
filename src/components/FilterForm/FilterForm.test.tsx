import { BrowserRouter } from "react-router-dom";
import { render } from "@testing-library/react";

import { DUMMY_FILTER_DATA, DUMMY_ORGANIZATION_DATA } from "../../constants";
import FilterForm from "./FilterForm";

describe("Filter Form Component Test", () => {
  it("Filter Form Component Snapshot", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <FilterForm
          filters={DUMMY_FILTER_DATA}
          setFilters={() => {}}
          runFilterFunction={() => {}}
          resetFilterFunction={() => {}}
          _organizations={DUMMY_ORGANIZATION_DATA}
        />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
