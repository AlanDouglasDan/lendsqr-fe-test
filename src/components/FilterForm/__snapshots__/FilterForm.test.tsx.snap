// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Form Component Test Filter Form Component Snapshot 1`] = `
<DocumentFragment>
  <div
    class="dropdown-menu p-4 form-dropdown"
  >
    <div
      class="mb-3"
    >
      <label
        class="form-label"
        for="exampleDropdownFormOrganization"
      >
        Organization
      </label>
      <select
        aria-label="Default select example"
        class="form-select mb-3 form-label"
        style="font-size: 1rem;"
      >
        <option>
          Select
        </option>
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
        <option />
      </select>
      <label
        class="form-label"
        for="exampleDropdownFormUsername"
      >
        Username
      </label>
      <input
        class="form-control"
        id="exampleDropdownFormUsername"
        placeholder="Username"
        type="text"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="form-label"
        for="exampleDropdownFormEmail2"
      >
        Email
      </label>
      <input
        class="form-control"
        id="exampleDropdownFormEmail2"
        placeholder="Email"
        type="email"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="form-label"
        for="datepicker"
      >
        Date
      </label>
      <input
        class="form-control"
        id="datepicker"
        type="date"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="form-label"
        for="exampleDropdownFormPhoneNumber"
      >
        Phone Number
      </label>
      <input
        class="form-control"
        id="exampleDropdownFormPhoneNumber"
        placeholder="Phone Number"
        type="text"
        value=""
      />
    </div>
    <label
      class="form-label"
      for="exampleDropdownFormStatus"
    >
      Status
    </label>
    <select
      aria-label="Default select example"
      class="form-select mb-3 form-label"
      style="font-size: 1rem;"
    >
      <option>
        Pending
      </option>
      <option
        value="active"
      >
        Active
      </option>
      <option
        value="inactive"
      >
        Inactive
      </option>
      <option
        value="blacklisted"
      >
        Blacklisted
      </option>
    </select>
    <div
      class="grid-2 mt-4"
    >
      <button
        class="btn-outline w-100"
      >
        Reset
      </button>
      <button
        class="btn-themed w-100"
      >
        Filter
      </button>
    </div>
  </div>
</DocumentFragment>
`;
